jax/_src/lax/parallel.py

```diff
@@ -157,6 +161,8 @@ def pmin(x, axis_name, *, axis_index_groups=None):
     Array(s) with the same shape as ``x`` representing the result of an
     all-reduce min along the axis ``axis_name``.
   """
+  if not isinstance(axis_name, (tuple, list)):
+    axis_name = (axis_name,)
```

Comment:
BTW one interesting issue with treating all tuples as collections of axes is that I once tried using a `taggedtuple` as a wrapper that would wrap an axis name and make it distinct form other axis names, but it later got unpacked in this code, because it satisfied `isinstance(x, tuple)`. Not necessary a comment to this PR, but a slightly weird behavior that might be surprising at times.

Reply:
Hrm interesting. Indeed having tuples represent collections of axis names is inconsistent with the description that anything hashable can be an axis name.

We could just make lists represent collections of axis names. But I think using lists and tuples interchangeably is so common in Python APIs that having them behave differently here would be confusing. So it seems okay to have tuples (and all subclasses) be an exception so that they act like collections rather than singletons.

Are there other options for how we could set things up?